src/fromager/settings.py

```diff
@@ -24,17 +28,35 @@ def packages(self) -> dict[str, dict[str, str]]:
             overrides.pkgname_to_override_module(key): value for key, value in p.items()
         }
 
-    def download_source_url(self, pkg: str, default: str | None = None) -> str | None:
+    def download_source_url(
+        self,
+        pkg: str,
+        req: Requirement | None = None,
+        version: Version | str | None = None,
+        default: str | None = None,
+        resolve_template: bool = True,
```

Comment:
Having the template rendering happen inside this module feels like a layering violation. It has to know what all the parameters allowed in the template might be, which means if we add parameters to the template we have to make changes in multiple places.

What if this method was `download_source_url_template(pkg: str)`? And the caller did the rendering? If the rendering logic is reusable, that could go into its own module.

Reply:
I put it in settings since templated values is something that settings provides to the user. I don't think the caller should be responsible or know how to do template resolution